src/osd/OSD.cc

```diff
@@ -9180,7 +9180,11 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb)
   osd->service.maybe_inject_dispatch_delay();
 
   boost::optional<PGQueueable> qi;
+
+  // we don't use a Mutex::Locker here because of the
+  // osd->service.release_reserved_pushes() call below
```

Comment:
Hmm, I guess this works — if somebody forgets to do an unlock it ought to become apparent pretty quickly. But if I ever have to touch this code I'll be setting up an RAII structure to call release_reserved_pushes when we need it. ;)